### PR TITLE
Activity log: Remove future days

### LINKED ISSUE #17695: Activity log: Don't show future days
We can see future days in the Activity Log. We should not show days after today:

extra-days

Related: #17678

### PR DESCRIPTION
**Pull request by sirreal**
#17695 modified the Activity Log month view to show all days in a month, including empty days.

This introduces a problem with the current month, which shows days in the _future_. This PR corrects that be comparing the end of the month with now.

## Testing
1. Visit https://calypso.live/stats/activity?branch=fix/activity-log/future-days
1. Do you see empty days in the current month? Do other months display correctly?

### Before

before

### After

after

Fixes #17695
cc/ @keoshi

### CODE REVIEW

**Comment by enejb** [comment]
I have tested this and it works as expected for me :shipit:

**Comment by sirreal** [comment]
I extracted `earliestMoment` to utils and added some tests.

@keoshi @rickybanister What should an empty **Today** look like? Currently it's not ideal, needs to be fixed up:

empty-today

**Comment by sirreal** [comment]
> No activity _yet_

@keoshi Do you mean `No activity <em>yet</em>`, or was the emphasis for clarity and not for html?

**Comment by keoshi** [comment]
@sirreal Sorry, for clarity only! :)

**Comment by sirreal** [comment]
no-activity-yet

File: client/my-sites/stats/activity-log-day/index.jsx [hunk 1]
```diff
@@ -115,6 +115,7 @@ class ActivityLogDay extends Component {
 		} = this.props;
 
 		const formattedDate = applySiteOffset( moment.utc( tsEndOfSiteDay ) ).format( 'LL' );
+		const noActivityText = isToday ? translate( 'No activity yet' ) : translate( 'No activity' );
```
Comment (a8ci18n) [inline client/my-sites/stats/activity-log-day/index.jsx]: Hi! I've found a possible matching string that has already been translated **84** times:
`translate( 'No activity yet!' )` *ES Score: 14*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/fix/activity-log/future-days)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

Comment (sirreal) [reply client/my-sites/stats/activity-log-day/index.jsx]: @keoshi how do you feel about reusing `'No activity yet!'` (with the exclamation)?

Comment (keoshi) [reply client/my-sites/stats/activity-log-day/index.jsx]: @sirreal missed this comment, sorry. I don't mind it, and if we're getting more consistency in the process better yet!

File: client/my-sites/stats/activity-log/utils.js [hunk 2]
```diff
@@ -24,3 +29,18 @@ export function adjustMoment( { timezone, gmtOffset, moment } ) {
...
+ *
+ * If moments are equal, return the first.
+ *
+ * @param  {...object} moments One or more moment objects
+ * @return {object}            Earliest moment from those provided
+ */
+export function earliestMoment( ...moments ) {
+	return reduce(
+		moments,
+		( current, compare ) => ( compare.valueOf() < current.valueOf() ? compare : current )
```
Comment (seear) [inline client/my-sites/stats/activity-log/utils.js]: Could maybe use `compare.isBefore( current )` here in place of the `<`.

File: client/my-sites/stats/activity-log/index.jsx [hunk 3]
```diff
@@ -263,7 +263,10 @@ class ActivityLog extends Component {
 
 		// loop backwards through each day in the month
 		for (
-			const m = this.applySiteOffset( moment.utc( startDate ) ).endOf( 'month' ).startOf( 'day' ),
+			const m = earliestMoment(
```
Comment (seear) [inline client/my-sites/stats/activity-log/index.jsx]: `math.min()` would work here I think in place of the custom `earliestMoment()`?

Comment (sirreal) [reply client/my-sites/stats/activity-log/index.jsx]: Turns out `moment` has `moment.min` with this exact call signature which I just discovered. Going to drop all of the redundant code and use 😅

Comment (seear) [reply client/my-sites/stats/activity-log/index.jsx]: :D

Comment (sirreal) [reply client/my-sites/stats/activity-log/index.jsx]: Cleaned up in e6b3dc8

File: client/my-sites/stats/activity-log-day/index.jsx (hunk 1 above)
Comment (sirreal) [reply client/my-sites/stats/activity-log-day/index.jsx]: f247a13 👍

**Comment by sirreal** [comment]
This should be all set to go now!

**Comment by seear** [comment]
Nice 👍

**Pull request merged**